Build with the Cython version that we have in ports.

Index: pyproject.toml
--- pyproject.toml.orig
+++ pyproject.toml
@@ -65,7 +65,7 @@ where = ["src"]
 "*" = ["*.c", "*.h", "*.pyx"]
 
 [build-system]
-requires = ["setuptools", "wheel", "pkgconfig", "Cython>=3", "setuptools_scm[toml]>=6.2"]
+requires = ["setuptools", "wheel", "pkgconfig", "Cython>=0.29.36", "setuptools_scm[toml]>=6.2"]
 build-backend = "setuptools.build_meta"
 
 [tool.setuptools_scm]
